crates/oxc_diagnostics/src/service.rs

```diff
@@ -128,17 +128,28 @@ impl DiagnosticService {
     /// Wrap [diagnostics] with the source code and path, converting them into [Error]s.
     ///
     /// [diagnostics]: OxcDiagnostic
-    pub fn wrap_diagnostics<P: AsRef<Path>>(
+    pub fn wrap_diagnostics<C: AsRef<Path>, P: AsRef<Path>>(
+        cwd: C,
         path: P,
         source_text: &str,
         source_start: u32,
         diagnostics: Vec<OxcDiagnostic>,
-    ) -> (PathBuf, Vec<Error>) {
-        let path = path.as_ref();
-        let path_display = path.to_string_lossy();
-        // replace windows \ path separator with posix style one
-        // reflects what eslint is outputting
-        let path_display = path_display.cow_replace('\\', "/");
+    ) -> Vec<Error> {
+        #[cfg(test)]
+        let is_jetbrains_terminal = false;
+        #[cfg(not(test))]
+        let is_jetbrains_terminal =
+            std::env::var("TERMINAL_EMULATOR").is_ok_and(|x| x.eq("JetBrains-JediTerm"));
+
+        let file_url = Url::from_file_path(path.as_ref());
```

Comment:
If there's a way to get the file URL without this dependency, it would probably be better to do it that way.